### DISCUSSION TITLE
why isn't `Greek` (or other scripts) listed in `UNICODE.md`? is the list out of date?

### DISCUSSION THREAD

**Discussion by BurntSushi** [rust-lang/regex#1144] [Q&A]
The documentation in [UNICODE.md](https://github.com/rust-lang/regex/blob/master/UNICODE.md#rl12-properties) includes a list of "all properties supported by the regex crate", but the list is missing "Greek", which is part of the examples in the documentation.
Is the list not updated? would you like the list to be generated from the implementation?

From @saona-raimundo: https://github.com/rust-lang/regex/issues/351#issuecomment-1875752784

**Comment by BurntSushi** [comment]
No, the list is not out of date. The `UNICODE.md` correctly lists "all properties supported by the regex crate." The confusion here is likely due to the fact that `Greek` is ***not*** a property. It's a property ***value*** for the `Script` property. It's for the same reason that `\p{Uppercase_Letter}` isn't listed there either. That's because `Uppercase_Letter` is not a property itself, but a property value for the `General_Category` property. The `UNICODE.md` file does not list all possible property values since these can technically change with new revisions of Unicode. (It's possible `General_Category` is special in that its property values will never change, but I can't remember the specific guarantees on this point.)

The reason why `\p{Greek}` works _as if_ `Greek` were a property is because `\p{SCRIPT_PROPERTY_VALUE}` and `\p{GENERAL_CATEGORY_PROPERTY_VALUE}` are both short-hands for `\p{sc:SCRIPT_PROPERTY_VALUE}` and `\p{gc:GENERAL_CATEGORY_PROPERTY_VALUE}` where `sc` and `gc` are the actual properties.

Unfortunately, Unicode support is rather complicated, and one cannot really understand it by just looking at the list of supported properties. The purpose of `UNICODE.md` is to document _which parts of [UTS#18](https://unicode.org/reports/tr18/#General_Category_Property) the `regex` crate conforms to_. To really understand Unicode support itself though, you really do need to read UTS#18.

If it were up to me, I wouldn't have made `\p{Greek}` a short-hand for `\p{sc:Greek}`. I'd make people type out `\p{sc:Greek}`. The reason for this is that it makes it look like `Greek` is just another property like anything else. But it isn't. It's a property value. It does still of course reference a finite set of codepoints, like all properties do, but it makes it a lot harder to look it up in the docs and figure out what it does. That is, because `\p{Greek}` looks a lot like `\p{Math}`, one might think that their definitions live in similar places or that they would be grouped together in the Unicode docs somewhere. But they aren't. They are categorically different things: `Greek` is a property value while `Math` is a property name.

**Reply by saona-raimundo** [reply]
Thank you! Now I understand!
